### PR TITLE
Default autoplay on for new installations only

### PR DESCRIPTION
**Pull request by mchowning**
## Description
This updates the auto play setting to only default to true for new installations.

## Testing Instructions

For all these tests, the up next queue needs to be empty.

### 1. Upgraded from having setting enabled
1. Install a build from this PR with the [version code](https://github.com/Automattic/pocket-casts-android/blob/dc7084f631248d0c65d5d37c1ad1b57d539c9ccc/base.gradle#L23) set to 1
2. Verify that the auto play setting is enabled
3. Upgrade the installation by Installing a build from this PR with the [version code](https://github.com/Automattic/pocket-casts-android/blob/dc7084f631248d0c65d5d37c1ad1b57d539c9ccc/base.gradle#L23) set to 2
4. Verify that the auto play setting is still enabled

### 2. Upgraded from having setting disabled
1. Install a build from this PR with the [version code](https://github.com/Automattic/pocket-casts-android/blob/dc7084f631248d0c65d5d37c1ad1b57d539c9ccc/base.gradle#L23) set to 1
2. Verify that the auto play setting is enabled
3. Change the setting to be disabled
4. Upgrade the installation by Installing a build from this PR with the [version code](https://github.com/Automattic/pocket-casts-android/blob/dc7084f631248d0c65d5d37c1ad1b57d539c9ccc/base.gradle#L23) set to 2
5. Verify that the auto play setting is still disabled

### 3. Upgraded from previous version of the app that did not have an auto play setting
1. Install a build from the `release/7.42` branch with the [version code](https://github.com/Automattic/pocket-casts-android/blob/dc7084f631248d0c65d5d37c1ad1b57d539c9ccc/base.gradle#L23) set to 1
2. Verify that there isn't an auto play setting in the settings
3. Upgrade the installation by Installing a build from this PR with the [version code](https://github.com/Automattic/pocket-casts-android/blob/dc7084f631248d0c65d5d37c1ad1b57d539c9ccc/base.gradle#L23) set to 2
5. Verify that the auto play setting is disabled

### 4. Watch
1. Install the app on a watch
2. Start playing an episode from a podcast with more than one podcast
3. Complete the episode
4. Verify that playback stops

### 5. Automotive
1. Install an Automtive build
2. Go to the app's settings
3. Verify that the auto play setting is on by default

## Checklist
- [X] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [X] I have considered whether it makes sense to add tests for my changes
- [X] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [X] Any jetpack compose components I added or changed are covered by compose previews

### CODE REVIEW

File: modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/AppLifecycleAnalytics.kt [hunk 1]
```diff
@@ -18,18 +18,18 @@ class AppLifecycleAnalytics @Inject constructor(
     /* The date the app was last opened, used for calculating time in app */
     private var applicationOpenedDate: Date? = null
 
-    fun onApplicationInstalledOrUpgraded() {
-        // Track app upgrade and install
-        val versionCode = packageUtil.getVersionCode(appContext)
-        val oldVersionCode = settings.getMigratedVersionCode()
```
Comment (ashiagr) [inline modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/AppLifecycleAnalytics.kt:24]: These are unused in the class constructor now and can be removed:

```
appContext: Context,
settings: Settings,
packageUtil: PackageUtil,
```

File: modules/features/shared/src/test/java/au/com/shiftyjelly/pocketcasts/shared/AppLifecycleObserverTest.kt [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import org.mockito.kotlin.spy
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+
+private const val VERSION_CODE_DEFAULT = 0
+private const val VERSION_CODE_AFTER_FIRST_INSTALL = 1
+private const val VERSION_CODE_AFTER_SECOND_INSTALL = 2
+
+@RunWith(MockitoJUnitRunner::class)
+class AppLifecycleObserverTest {
```
Comment (ashiagr) [inline modules/features/shared/src/test/java/au/com/shiftyjelly/pocketcasts/shared/AppLifecycleObserverTest.kt:31]: Kudos for adding these tests 🙌

**Review by ashiagr — APPROVED** [review]
Everything works great! Just left a nitpick.

File: modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/AppLifecycleAnalytics.kt (hunk 1 above)
Comment (mchowning) [reply modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/AppLifecycleAnalytics.kt:24]: Thanks for catching that. Fixed in [68daf84](https://github.com/Automattic/pocket-casts-android/pull/1134/commits/68daf84546bb01469e93bf4ea9d5d03122d3c6b6)

**Pull request merged**